osh/builtin_assign.py

```diff
@@ -25,10 +26,108 @@
   from frontend import arg_def
 
 
+def _PrintVariables(mem, cmd_val, arg, print_flags, readonly = False, exported = False):
+  flag_g = getattr(arg, 'g', None)
```

Comment:
These can all just be `arg.g`, `arg.n`, etc. right?  And then you don't really need the temporary

Reply:
This function `_PrintVariables` is shared by `class Export`, `Readonly`, and `NewVar`. Each has its own definition for `arg`. `Export`, `Readonly`, and `NewVar` supports flags `-n`, `-aA`, and `-gnrxaA`, respectively. For example, some have flag `g` and the others don't have the flag `g`. So, one first needs to check whether each attribute is defined in `arg` or not, or otherwise, I get `AttributeError`.

I thought about one possibility to create a dictionary at the caller side something like `_PrintVariables(..., arg={'g': None, 'n': None, 'r': None, 'x': None, 'a': arg.a, 'A': arg.A}, ...)`. But it looks like redundant codes. Also, we need to keep consistency between this function call and the definitions of `arg` of each classes; every time one updates the set of flags in `arg`, one needs to update this function call accordingly, which is not so maintainable.

Another possibility may be to support the same set of flags for `export` and `readonly` as `declare`.

Do you have an idea of a more clever way to implement this?

Reply:
Ah OK I see.   This is OK for now but could use a comment saying that.

(I think I have an idea of how it will translate to C++ ... the dynamic types of `arg.X` are an issue I've been thinking about.)

Reply:
OK!

> This is OK for now but could use a comment saying that.

It has been already merged, so could you write a comment for this? Thank you!